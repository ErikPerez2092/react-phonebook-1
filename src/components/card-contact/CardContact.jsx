import React from 'react';
import PropTypes from 'prop-types';
import './cardContact.css';

const CardContact = ({ img, name, nickname, group }) => (
  <div className="CardContact">
    <div className="CardContact-img" style={{ backgroundImage: `url(${img})` }} />
    <h2 className="CardContact-name">{name}</h2>
    <h3 className="CardContact-nickname">{nickname}</h3>
    <span className="CardContact-group">{group}</span>
  </div>
);

CardContact.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  nickname: PropTypes.string.isRequired,
  group: PropTypes.string.isRequired,
};

export default CardContact;
