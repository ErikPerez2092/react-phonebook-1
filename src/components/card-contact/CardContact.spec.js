import React from 'react';
import { shallow } from 'enzyme';
import CardContact from './CardContact';

describe('CardContactComponent', () => {
  let cardContactComponent;
  beforeEach(() => {
    cardContactComponent = shallow(<CardContact img="https://randomuser.me/api/portraits/men/83.jpg" name="Edgar" nickname="El Zavalia" group="Friends" />);
  });

  it('sould render <CardContact />', () => {
    expect(cardContactComponent.getElements()).toMatchSnapshot();
  });

  it('should contain an image', () => {
    expect(cardContactComponent.find('.CardContact-img')).toHaveLength(1);
  });

  it('should have a name', () => {
    expect(cardContactComponent.find('h2')).toHaveLength(1);
    expect(cardContactComponent.find('h2').text()).toBe('Edgar');
  });
});
