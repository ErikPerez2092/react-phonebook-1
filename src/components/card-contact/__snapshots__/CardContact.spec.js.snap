// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContactComponent sould render <CardContact /> 1`] = `
Array [
  <div
    className="CardContact"
>
    <div
        className="CardContact-img"
        style={
            Object {
                "backgroundImage": "url(https://randomuser.me/api/portraits/men/83.jpg)",
              }
        }
    />
    <h2
        className="CardContact-name"
    >
        Edgar
    </h2>
    <h3
        className="CardContact-nickname"
    >
        El Zavalia
    </h3>
    <span
        className="CardContact-group"
    >
        Friends
    </span>
</div>,
]
`;
